metrics/metric_observer_test.go

```diff
@@ -1,19 +1,135 @@
 package metrics
 
 import (
+	"bytes"
 	"testing"
 
+	"github.com/openshift-online/async-routine"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/testutil"
+	"go.uber.org/mock/gomock"
 )
 
 func TestPrometheusMetrics(t *testing.T) {
```

Comment:
@ziccardi @simonpasquier what's the effort of transofrming this test to ginkgo?

Reply:
not much I guess, I'll update.